commit c63b8ad0aeb7c83671c1cdde3ceb65b99edaf78b
Author: aszlig <aszlig@nix.build>
Date:   Fri Feb 15 20:41:54 2019 +0100

    Disable user id check for config.php
    
    Nextcloud checks whether the user matches the webserver user by
    comparing the current userid with the owner of config.php. In our case
    however, the config.php is inside the Nix store so it most certainly
    isn't owned by the nextcloud user.
    
    Signed-off-by: aszlig <aszlig@nix.build>

diff --git a/console.php b/console.php
index 88456a9bd7..ee570faa75 100644
--- a/console.php
+++ b/console.php
@@ -60,21 +60,6 @@ try {
 
 	set_exception_handler('exceptionHandler');
 
-	if (!function_exists('posix_getuid')) {
-		echo "The posix extensions are required - see https://www.php.net/manual/en/book.posix.php" . PHP_EOL;
-		exit(1);
-	}
-	$user = posix_getuid();
-	$configUser = fileowner(OC::$configDir . 'config.php');
-	if ($user !== $configUser) {
-		echo "Console has to be executed with the user that owns the file config/config.php" . PHP_EOL;
-		echo "Current user id: " . $user . PHP_EOL;
-		echo "Owner id of config.php: " . $configUser . PHP_EOL;
-		echo "Try adding 'sudo -u #" . $configUser . "' to the beginning of the command (without the single quotes)" . PHP_EOL;
-		echo "If running with 'docker exec' try adding the option '-u " . $configUser . "' to the docker command (without the single quotes)" . PHP_EOL;
-		exit(1);
-	}
-
 	$oldWorkingDir = getcwd();
 	if ($oldWorkingDir === false) {
 		echo "This script can be run from the Nextcloud root directory only." . PHP_EOL;
diff --git a/cron.php b/cron.php
index ffce18b98d..5da6c32d2c 100644
--- a/cron.php
+++ b/cron.php
@@ -89,22 +89,6 @@ try {
 			@set_time_limit(0);
 		}
 
-		// the cron job must be executed with the right user
-		if (!function_exists('posix_getuid')) {
-			echo "The posix extensions are required - see https://www.php.net/manual/en/book.posix.php" . PHP_EOL;
-			exit(1);
-		}
-
-		$user = posix_getuid();
-		$configUser = fileowner(OC::$configDir . 'config.php');
-		if ($user !== $configUser) {
-			echo "Console has to be executed with the user that owns the file config/config.php" . PHP_EOL;
-			echo "Current user id: " . $user . PHP_EOL;
-			echo "Owner id of config.php: " . $configUser . PHP_EOL;
-			exit(1);
-		}
-
-
 		// We call Nextcloud from the CLI (aka cron)
 		if ($appMode !== 'cron') {
 			$config->setAppValue('core', 'backgroundjobs_mode', 'cron');
